### ISSUE TITLE
[docs] The relation between source type and rendering type needs to be clarified.

### ISSUE THREAD

**Issue by jingsam** [mapbox/mapbox-gl-js#5729]
In mapbox-style-spec, we have `source types` as:
```
vector
geojson
raster
image
video
canvas
```

And then we have `rendering type` as:
```
fill
fill-extrusion
line
symbol
circle
heatmap
raster
```

However, the docs did not say a souce type can be rendered as wich rending type, or a rendering type should specify which type of source. For example, how to add a layer that have a image\video\canvas type.

I conclude as below:
![image](https://user-images.githubusercontent.com/1522494/33059052-9228ff74-cecd-11e7-8b78-e3d3264edf2e.png)

I think the docs should clarify this relation.

**Comment by 1ec5** [comment]
Yes, this would be wonderful, especially the part mapping geometry types to style layers. That would’ve helped to head off the confusion in #4002.

The iOS SDK does contain this documentation, albeit only in plain text. If we derive the additional documentation from new fields in the style specification JSON, then [these iOS SDK overrides](https://github.com/mapbox/mapbox-gl-native/blob/d46641cad61c2f7e634eae6383d30a8b13cd43a9/platform/darwin/scripts/style-spec-overrides-v8.json#L11-L30) can get a bit shorter.
